libraries/AP_Param/AP_Param.h

```diff
@@ -887,6 +887,28 @@ class AP_ParamV : public AP_Param
         save(force);
     }
 
+    /// Combined set and save, but only does the save if the value is
+    /// different from the current ram value, thus saving us a
+    /// scan(). This should only be used where we have not set() the
+    /// value separately, as otherwise the value in EEPROM won't be
+    /// updated correctly.
+    void set_and_save_ifchanged(const T &v) {
+        if (vtype != AP_PARAM_VECTOR3F) {
```

Comment:
this is inside the ParamV class, how can it not be a vector?

Reply:
that's right will get rid of this.